xl-objects.go

```diff
@@ -190,9 +190,11 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64) (io.Read
 			// Reset offset to 0 as it would be non-0 only for the first loop if startOffset is non-0.
 			offset = 0
 			if _, err := io.Copy(fileWriter, r); err != nil {
+				r.Close()
```

Comment:
> +                               r.(*io.PipeReader).CloseWithError(err)

To avoid possible problems of type changes, use 

```
pr, ok :=   r.(*io.PipeReader)
if ok {
   pr.CloseWithError(err)
```